backend/routes/users.js

```diff
@@ -0,0 +1,121 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const keys = require("../config/keys");
+const passport = require("passport");
+
+// Load input validation
+const validateRegisterInput = require("../validation/signUp");
+const validateLoginInput = require("../validation/signIn");
+
+// Load User model
+const User = require("../models/user");
+
+const axios = require("axios");
+
+router.post("/insertOne", (req, res) => {
+  // Form validation
+  console.log("Received signup request");
+  
+  const { errors, isValid } = validateRegisterInput(req.body.document);
+
+  // Check validation
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+  
+  User.findOne({ email: req.body.document.email }).then(user => {
+    if (user) {
+      return res.status(400).json({ email: "Email already exists" });
```

Comment:
Should it be the `email` field that contains the error message `"Email already exists"` or should it be a field named something like `errorMsg`?

Reply:
This was an oversight on my part. I am adding this error handeling into validation/signUp.js instead.